x/group/keeper/msg_server.go

```diff
@@ -345,12 +344,8 @@ func (k Keeper) CreateGroupPolicy(goCtx context.Context, req *group.MsgCreateGro
 			// next sequence value and derive a new address.
 			continue
 		}
-		acc := k.accKeeper.NewAccount(ctx, &authtypes.ModuleAccount{
-			BaseAccount: &authtypes.BaseAccount{
-				Address: accountAddr.String(),
-			},
-			Name: accountAddr.String(),
-		})
+
+		acc := k.accKeeper.NewAccountWithAddress(ctx, accountAddr)
```

Comment:
This won't set a module account name for a group policy, as we cannot derive a group policy address from a group policy address (breaking the validate-genesis).
This is state machine breaking.